src/components/base/Dialogs/AlertDialog.tsx

```diff
@@ -13,18 +13,47 @@ import {
   Text,
 } from "@chakra-ui/react";
 import { SwapState } from "../../Swap";
-import { CloseIcon } from "@chakra-ui/icons";
+import { CloseIcon, WarningIcon } from "@chakra-ui/icons";
+import { LoadingDialog } from "./LoadingDialog";
 
+export enum SwapConfirmationStep {
+  CONFIRM, // Initial step/state
+  SIGN, // When hashpack is awaiting signature
+  TRANSACTION, // When transaction is signed in hashpack and is now executing (NOTE: currently cannot support)
+  ERROR, // When there is an error with the executed transaction (for error dialog)
+}
 interface SwapConfirmationProps {
   sendSwapTransaction: (payload: any) => void;
   swapState: SwapState;
+  confirmationStep: SwapConfirmationStep;
+  errorMessage?: string;
+  onErrorMessageDismiss?: () => void | undefined; // fired when error dialog is closed (SwapConfirmationStep.ERROR)
+  onSwapButtonClick?: () => void | undefined; // fired when swap button is clicked (button starts dialog flow)
+  onClose?: () => void | undefined; // fired when swap setting AlertDialog is closed (sendSwapTransaction)
 }
 
 const SwapConfirmation = (props: SwapConfirmationProps) => {
```

Comment:
It would be best if we can decouple the swap logic and props from the AlertDialog component. The AlertDialog component is intended to be a simple component that can be used as a base to construct domain specific dialogs modals.